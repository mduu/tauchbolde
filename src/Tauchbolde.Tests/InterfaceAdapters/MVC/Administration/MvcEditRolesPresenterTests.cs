using System;
using System.Collections.Generic;
using ApprovalTests.Reporters;
using FluentAssertions;
using Tauchbolde.Application.UseCases.Administration.GetEditRolesUseCase;
using Tauchbolde.InterfaceAdapters.MVC.Presenters.Administration.EditRoles;
using Tauchbolde.Tests.TestingTools;
using Tauchbolde.Tests.TestingTools.TestDataFactories;
using Xunit;

namespace Tauchbolde.Tests.InterfaceAdapters.MVC.Administration
{
    [UseReporter(typeof(DiffReporter))]
    public class MvcEditRolesPresenterTests
    {
        private readonly MvcEditRolesPresenter presenter = new MvcEditRolesPresenter();

        [Fact]
        public void Output_Success()
        {
            // Arrange
            var interactorOutput = CreateValidInteractorOutput();

            // Act
            presenter.Output(interactorOutput);
            var viewModel = presenter.GetViewModel();

            // Assert
            ApprovalsExtensions.VerifyObjectJson(viewModel);
        }

        [Fact]
        public void Output_NullOutput_MustThrow()
        {
            // Act
            // ReSharper disable once AssignNullToNotNullAttribute
            Action act = () => presenter.Output(null);
            
            // Assert
            act.Should().Throw<ArgumentNullException>().Which.ParamName.Should().Be("interactorOutput");
        }

        private GetEditRolesOutput CreateValidInteractorOutput() => 
            new GetEditRolesOutput(
                DiverFactory.JohnDoeUserName,
                DiverFactory.JohnDoeFullname,
                new List<string> { "role1", "role2," }, 
                new List<string> { "role1" });
    }
}