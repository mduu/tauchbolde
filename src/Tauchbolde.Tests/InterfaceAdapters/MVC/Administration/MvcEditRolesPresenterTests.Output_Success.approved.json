{
    "UserName": "john.doe",
    "FullName": "John Doe",
    "AllRoles": [
        "role1",
        "role2,"
    ],
    "AssignedRoles": [
        "role1"
    ]
}